### ISSUE TITLE
[Feature Request]: option to generate smaller thumbnails for images above a certain size

### ISSUE THREAD

**Issue by BPplays** [TagStudioDev/TagStudio#345]
### Checklist

- [X] I am using an up-to-date version.
- [X] I have read the [documentation](https://github.com/TagStudioDev/TagStudio/blob/main/doc/index.md).
- [X] I have searched existing [issues](https://github.com/TagStudioDev/TagStudio/issues).

### Description

sometimes large images can take a relatively long time to load

### Solution

an option to have TagStudio generate smaller thumbnails for images above a certain size in configurable sizes, formats (e.g. jxl, webp, png, etc), and quality that are loaded first before the main image would improve responsiveness while navigating images

### Alternatives

some formats support progressive decoding but jxl is the only one i can think of that could act in a similar way but even then not all jxls support progressive decoding.
also I'm not sure if jxl progressive decoding would fix this issue, I'm not sure what makes up the load time for images if it's primarily disc usage or rendering

**Comment by CyanVoxel** [comment]
I don't believe generating smaller resulting images for large images would improve rendering times, as the entire file would still need to be parsed and re-rendered (outside of anything like progressive decoding, but the vast majority of images are not .jxl).

A couple alternative solutions planned to help with this are thumbnail caching and loading the rest of the preview panel information while the previews render separately.

**Comment by BPplays** [comment]
@CyanVoxel
i might have explained what i want poorly, i meant i want to generate lower res images and have them saved to disk to be loaded first later.
